src/token.rs

```diff
@@ -34,29 +32,29 @@ impl Token {
     }
 
     // TODO: should not need to use mem::take and then place back when the token is valid
-    pub async fn get(&mut self) -> crate::Result<&str> {
+    pub async fn get(&mut self, client: &reqwest::Client) -> crate::Result<&str> {
         match std::mem::take(&mut self.token) {
             Some((token, exp)) if exp > now() => {
                 self.token = Some((token, exp));
                 Ok(&self.token.as_ref().unwrap().0)
             }
-            _ => self.retrieve().await,
+            _ => self.retrieve(client).await,
         }
     }
 
-    async fn retrieve(&mut self) -> crate::Result<&str> {
-        self.token = Some(Self::get_token(&self.access_scope).await?);
+    async fn retrieve(&mut self, client: &reqwest::Client) -> crate::Result<&str> {
+        self.token = Some(Self::get_token(client, &self.access_scope).await?);
         match self.token {
             Some(ref token) => Ok(&token.0),
             None => unreachable!(),
         }
     }
 
-    async fn get_token(scope: &str) -> Result<(String, u64), Error> {
-        let now = now();
+    async fn get_token(client: &reqwest::Client, scope: &str) -> crate::Result<(String, u64)> {
+        let now = crate::token::now();
```

Comment:
Why are these paths fully specified, instead of just writing `now()`?

Reply:
Huh, very strange. I might have moved some code around (needing to qualify it) and then moved it back. Fixed.

Reply:
Rust-analyzer to the rescue I'm sure :P